projects/netgrif-components-core/src/lib/forms/change-password/abstract-change-password.component.ts

```diff
@@ -0,0 +1,107 @@
+import {passwordValidator} from '../models/password.validator';
+import {LoggerService} from '../../logger/services/logger.service';
+import {FormBuilder, FormGroup, Validators} from '@angular/forms';
+import {MessageResource} from '../../resources/interface/message-resource';
+import {TranslateService} from '@ngx-translate/core';
+import {Component, EventEmitter, OnDestroy, Output} from '@angular/core';
+import {FormSubmitEvent, HasForm} from "../has-form";
+import {LoadingEmitter} from "../../utility/loading-emitter";
+import {take} from "rxjs/operators";
+import {UserChangePasswordRequest} from "../../authentication/profile/models/user-change-password-request";
+import {ProfileService} from "../../authentication/profile/services/profile.service";
+import {UserService} from "../../user/services/user.service";
+import {encodeBase64} from '../../utility/base64';
+
+@Component({
+    selector: 'ncc-abstract-forgotten-password',
+    template: ''
+})
+export abstract class AbstractChangePasswordComponent implements HasForm, OnDestroy {
+
+    protected readonly MIN_PASSWORD_LENGTH = 8;
```

Comment:
should probably be a global constant, not a class specific one

Reply:
Done InjectToken